zebra-consensus/src/block/check.rs

```diff
@@ -246,9 +260,19 @@ pub fn miner_fees_are_valid(
     // https://zips.z.cash/protocol/protocol.pdf#txnconsensus
     let left = (transparent_value_balance - sapling_value_balance - orchard_value_balance)
         .map_err(|_| SubsidyError::SumOverflow)?;
-    let right = (block_subsidy + block_miner_fees).map_err(|_| SubsidyError::SumOverflow)?;
+    let right = (block_subsidy + block_miner_fees - expected_deferred_amount)
+        .map_err(|_| SubsidyError::SumOverflow)?;
+
+    // TODO: Add link to exact coinbase balance ZIP
+    let should_allow_unclaimed_subsidy =
+        NetworkUpgrade::current(network, height) <= NetworkUpgrade::Nu5;
```

Comment:
Although this is probably fine in practice, I note that it *assumes* that whatever upgrade is next after NU5 will activate the ZIP for https://zips.z.cash/draft-hopwood-coinbase-balance . I suggest gating `should_allow_unclaimed_subsidy` specifically on NU6 not being activated.

Reply:
> I suggest gating should_allow_unclaimed_subsidy specifically on NU6 not being activated.

Applied in https://github.com/ZcashFoundation/zebra/pull/8727/commits/993e915e89a59594b6a2faa581b5dc62bd1b9816.